packages/wdio-browserstack-service/src/reporter.ts

```diff
@@ -0,0 +1,83 @@
+import WDIOReporter, { SuiteStats, TestStats, RunnerStats } from '@wdio/reporter'
+import type { Capabilities, Options } from '@wdio/types'
+
+import { v4 as uuidv4 } from 'uuid'
+import { Browser, MultiRemoteBrowser } from 'webdriverio'
+
+import { BrowserstackConfig, TestData } from './types'
+import { getCloudProvider, uploadEventData, getHierarchy } from './util'
+import RequestQueueHandler from './request-handler'
+
+export default class TestReporter extends WDIOReporter {
+    private _capabilities: Capabilities.Capabilities = {}
+    private _config?: BrowserstackConfig & Options.Testrunner
+    private _observability = true
+    private _sessionId?: string
+    private _suiteName?: string
+    private _requestQueueHandler = RequestQueueHandler.getInstance()
+
+    onRunnerStart (runnerStats: RunnerStats) {
+        this._capabilities = runnerStats.capabilities as Capabilities.Capabilities
+        this._config = runnerStats.config as BrowserstackConfig & Options.Testrunner
+        this._sessionId = runnerStats.sessionId
+        /* istanbul ignore next */
+        if (this._config.testObservability == false) this._observability = false
```

Comment:
Never use `==` comparisons
```suggestion
        this._observability = this._config.testObservability
```